### PR TITLE
Fix search of relative assets in derived nodes

### PR DESCRIPTION
**Pull request by ad-daniel**
**Description**
Fixes: https://github.com/cyberbotics/webots/issues/5031

Assets relative to a PROTO are determined based on the visibility of the field in question. For nested PROTO, following the aliases allows to determine the scope of a field and therefore relative to which PROTO it must be resolved.
This method however falls flat when derived PROTO are concerned because in the process of building these PROTO nodes, all intermediary nodes are instantiated only temporarily, so notions of visibility can't be relied upon as no contiguous chain of aliases exists anymore.

With this PR, a map of "IS" connections (between the exposed parameter and the internal field) is kept at the PROTO model level so that it's possible to follow this chain by jumping through consecutive ancestors in the derived PROTO chain.

### CODE REVIEW

**Comment by BenjaminDeleze** [comment]
With a bit of luck it will also fix #5028 !

**Comment by ad-daniel** [comment]
> With a bit of luck it will also fix #5028 !

eheh, no such luck unfortunately, this issue/fix is just related to derived PROTO, it doesn't touch the visibility condition

File: src/webots/vrml/WbProtoModel.hpp [hunk 1]
```diff
@@ -145,9 +147,11 @@ class WbProtoModel : public QObject {
   QStringList mTags;
   QString mTemplateLanguage;
 
+  QMap<QString, QString> mParameterAliases;  // tracks the connections between an exposed parameter and its internal counterpart
+
   ~WbProtoModel();  // called from unref()
-  void verifyAliasing(WbNode *root, WbTokenizer *tokenizer) const;
-  void verifyNodeAliasing(WbNode *node, WbFieldModel *param, WbTokenizer *tokenizer, bool searchInParameters, bool &ok) const;
+  void verifyAliasing(WbNode *root, WbTokenizer *tokenizer);
+  void verifyNodeAliasing(WbNode *node, WbFieldModel *param, WbTokenizer *tokenizer, bool searchInParameters, bool &ok);
```
Comment (omichel) [inline src/webots/vrml/WbProtoModel.hpp:154]: I would rename these functions to `setupAliasing` and `setupNodeAliasing` as the `verify` prefix is a bit misleading (they also construct the parameter alias map).

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**